pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/notPepperKit/NotPepperKitCreationService.java

```diff
@@ -0,0 +1,238 @@
+            return false;
+        }
+        //let's validate the participant's address
+        String name = "";
+        if (StringUtils.isNotBlank(juniperKitRequest.getFirstName())) {
+            name += juniperKitRequest.getFirstName() + " ";
+        }
+        name += juniperKitRequest.getLastName();
+        if (juniperKitRequest.isSkipAddressValidation()) {
+            try {
+                Address address = easyPostUtil.createBroadAddress(name, juniperKitRequest.getStreet1(), juniperKitRequest.getStreet2(), juniperKitRequest.getCity(),
+                        juniperKitRequest.getPostalCode(), juniperKitRequest.getState(), juniperKitRequest.getCountry(), phone);
+                juniperKitRequest.setEasypostAddressId(address.getId());
+                return true;
+            } catch (EasyPostException e) {
+                log.error("Easypost couldn't create an address for " + juniperKitRequest.getShortId());
+                return false;
+            }
+        }
+        DeliveryAddress deliveryAddress =
+                new DeliveryAddress(juniperKitRequest.getStreet1(), juniperKitRequest.getStreet2(), juniperKitRequest.getCity(), juniperKitRequest.getState(),
+                        juniperKitRequest.getPostalCode(), juniperKitRequest.getCountry(), name, phone);
+        deliveryAddress.validate();
+        if (deliveryAddress.isValid()) {
+            //store the address back
+            juniperKitRequest.setEasypostAddressId(deliveryAddress.getId());
+            return true;
+        }
+        log.info("Address is not valid " + juniperKitRequest.getShortId());
+        return false;
+
+    }
+
+
+    /**
+     * This method creates a collaborator participant id for the kit and then inserts in DB
+     */
+    private void createKit(@NonNull DDPInstance ddpInstance, @NonNull KitType kitType, JuniperKitRequest kit,
+                           @NonNull KitRequestSettings kitRequestSettings, @NonNull EasyPostUtil easyPostUtil, @NonNull String kitTypeName,
+                           ArrayList<KitRequest> orderKits, Connection conn) {
+
+        String externalOrderNumber = null;
+        if (StringUtils.isNotBlank(kitRequestSettings.getExternalShipper())) {
+            externalOrderNumber = DDPKitRequest.generateExternalOrderNumber();
+        }
+        String errorMessage = "";
+        String collaboratorParticipantId = KitRequestShipping
+                .getCollaboratorParticipantId(null, ddpInstance.getDdpInstanceId(), ddpInstance.isMigratedDDP(),
+                        ddpInstance.getCollaboratorIdPrefix(), kit.getJuniperParticipantID(), kit.getJuniperParticipantID(),
+                        kitRequestSettings.getCollaboratorParticipantLengthOverwrite());
+
+        //In case it's needed, subkits handling should be added here
+        String shippingId;
+        String userId;
+        //checking ddpInstance.isHasRole() to know this is a Juniper Kit
```

Comment:
It seems like we already did this checking elsewhere.
If so, perhaps something more like an assertion at the beginning of the method is needed (a check that throws a 500 error).